source/Octopus.Server.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -81,20 +81,23 @@ private async Task AssertSpaceIdMatchesResource(TResource resource)
                 await CheckSpaceResource(spaceResource).ConfigureAwait(false);
         }
 
-        protected async Task<bool> ThrowIfServerVersionIsNotCompatible()
+        // TODO: default value on token is a stop-gap until cancellation token is fully supported across. 
+        protected async Task<bool> ThrowIfServerVersionIsNotCompatible(CancellationToken cancellationToken = default)
```

Comment:
This is a base class for many `concrete` repository. As per previous comment on full cancellation token support, this default value will be redundant then.